pkg/operator2/route.go

```diff
@@ -22,7 +22,14 @@ func (c *authOperator) handleRoute(ingress *configv1.Ingress) (*routev1.Route, *
 	if err != nil {
 		return nil, nil, "FailedCreate", err
 	}
+	defRoute := defaultRoute(ingress)
 
+	if route.Spec.Host != defRoute.Spec.Host {
+		route, err = c.route.Update(defaultRoute(ingress))
```

Comment:
Note that this approach to update will fail because resource version will be unset.